src/modules/forking/components/migrate-rep-form/migrate-rep-form.jsx

```diff
@@ -0,0 +1,258 @@
+/* eslint jsx-a11y/label-has-for: 0 */
+
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import classNames from 'classnames'
+import BigNumber from 'bignumber.js'
+
+import { BINARY, SCALAR } from 'modules/markets/constants/market-types'
+import { ExclamationCircle as InputErrorIcon } from 'modules/common/components/icons'
+import FormStyles from 'modules/common/less/form'
+import Styles from 'modules/forking/components/migrate-rep-form/migrate-rep-form.styles'
+
+export default class MigrateRepForm extends Component {
+
+  static propTypes = {
+    market: PropTypes.object.isRequired,
+    updateState: PropTypes.func.isRequired,
+    validations: PropTypes.object.isRequired,
+    selectedOutcome: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
+    selectedOutcomeName: PropTypes.string.isRequired,
+    forkMigrationTotals: PropTypes.object.isRequired,
+    repAmount: PropTypes.number,
+    isMarketInValid: PropTypes.bool,
+  }
+
+  static checkRepAmount(repAmount, updatedValidations) {
+    if (repAmount === '' || repAmount == null || repAmount <= 0) {
+      updatedValidations.repAmount = 'The rep Amount field is required.'
+    } else {
+      delete updatedValidations.repAmount
+    }
+    return updatedValidations
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      outcomes: [],
+      inputRepAmount: '',
+      inputSelectedOutcome: '',
+      maxRep: 0,
+    }
+
+    // TODO Reportable outcomes?
+    this.state.outcomes = this.props.market ? this.props.market.outcomes.slice() : []
+    if (this.props.market && this.props.market.marketType === BINARY && this.props.market.outcomes.length === 1) {
+      this.state.outcomes.push({ id: 0, name: 'No' })
+    }
+
+    this.state.outcomes.sort((a, b) => a.name - b.name)
+    if (this.props.repAmount) this.state.inputRepAmount = this.props.repAmount.toString()
+
+    this.componentWillReceiveProps(this.props)
+
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.selectedOutcome || typeof nextProps.selectedOutcome === 'number') {
+      if (!this.state.outcomes.find(o => o.id === nextProps.selectedOutcome)) {
+        this.state.inputSelectedOutcome = nextProps.selectedOutcome
```

Comment:
Use this.setState here. See: https://reactjs.org/docs/react-component.html#componentwillreceiveprops

Reply:
....and given you are calling this from the constructor, react may not be happy.